cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -59,12 +56,16 @@ set(ARROW_THIRDPARTY_DEPENDENCIES
   BOOST)
 
 # TODO: double-conversion check fails for conda, it should not
-if(ARROW_DEPENDENCY_SOURCE STREQUAL "CONDA")
+if (ARROW_DEPENDENCY_SOURCE STREQUAL "CONDA")
   message(STATUS "Using CONDA_PREFIX: $ENV{CONDA_PREFIX}")
   set(ARROW_PACKAGE_PREFIX $ENV{CONDA_PREFIX})
   set(ARROW_ACTUAL_DEPENDENCY_SOURCE "SYSTEM")
+else()
+  set(ARROW_ACTUAL_DEPENDENCY_SOURCE "${ARROW_DEPENDENCY_SOURCE}")
 endif()
 
+message(STATUS "Using ${ARROW_ACTUAL_DEPENDENCY_SOURCE} approach to find dependencies")
```

Comment:
Moving this message here, we should also add a messge when the `CONDA_PREFIX` was applied.